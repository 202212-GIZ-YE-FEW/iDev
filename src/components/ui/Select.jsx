import clsx from "clsx";

function Select(props) {
    const {
        label,
        placeholder,
        value,
        options,
        labelColor = "light-gray",
        fontSize = "xl",
        fontWeight = "light",
        widthSize = "w-full",
        heightSize = "h-auto",
        radius = "md",
        shadow = "sm",
        onChange,
        className = "",
    } = props;

    const inputClasses = clsx(
        `${widthSize} ${heightSize} border text-base text-light-black border-solid border-light-gray/30 shadow-${shadow} self-center placeholder-light-gray leading-[2.15] px-[20px] py-[10px] min-w-0 text-${widthSize} font-${fontWeight} rounded-${radius} ${className}`,
        {
            [`text-${fontSize}`]: heightSize === "h-auto",
        }
    );

    return (
        <>
            {label && (
                <label
                    className={`mt-1 mb-2 whitespace-wrap text-sm md:text-base lg:text-lg flex:me-10 text-${labelColor} font-light flex:self-center capitalize flex-1`}
                >
                    {label}
                </label>
            )}

            <select
                value={value}
                onChange={onChange}
                placeholder={placeholder}
                className={inputClasses}
            >
                {options?.map((item, key) => {
                    return (
                        <option key={key} value={item.value}>
                            {item.label}
                        </option>
                    );
                })}
            </select>
        </>
    );
}

export default Select;
