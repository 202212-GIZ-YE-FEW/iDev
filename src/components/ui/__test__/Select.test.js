import renderer from "react-test-renderer";

import Select from "../Select";

it("renders correctly", () => {
    const tree = renderer.create(<Select />).toJSON();
    expect(tree).toMatchSnapshot();
});
