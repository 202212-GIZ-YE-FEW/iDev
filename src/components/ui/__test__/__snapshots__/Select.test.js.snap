// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<select
  className="w-full h-auto block border text-sm border-solid self-center text-light-gray border-light-gray/30 px-[10px] min-w-0 font-light rounded-md shadow-sm flex-[2_1_0%]  text-xl"
  onChange={[Function]}
/>
`;
